src/net/sourceforge/kolmafia/persistence/SkillDatabase.java

```diff
@@ -161,11 +93,63 @@ public static final int skillTypeNameToType(final String typeName) {
         SkillDatabase.GNOME_SKILLS,
         SkillDatabase.BAD_MOON
       };
+  private static final int[] NO_SKILL_IDS = new int[0];
+  private static final AdventureResult SUPER_SKILL = EffectPool.get(EffectPool.SUPER_SKILL);
+  private static final ArrayList<String> skillNames = new ArrayList<>();
+  private static String[] canonicalNames = new String[0];
 
   static {
     SkillDatabase.reset();
   }
 
+  public static final String skillTypeToTypeName(final int type) {
+    return type == PASSIVE
```

Comment:
(I would like to rip out this and the next function and replace them with switch statements and/or expressions, but best to deal with that separately.)